src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarBattleCommanderUtil.java

```diff
@@ -0,0 +1,64 @@
+package com.gmail.goosius.siegewar.utils;
+
+import com.gmail.goosius.siegewar.SiegeController;
+import com.gmail.goosius.siegewar.enums.SiegeSide;
+import com.gmail.goosius.siegewar.enums.SiegeWarPermissionNodes;
+import com.gmail.goosius.siegewar.objects.Siege;
+import com.gmail.goosius.siegewar.settings.SiegeWarSettings;
+import com.palmergames.bukkit.towny.TownyAPI;
+import com.palmergames.bukkit.towny.object.Translatable;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+
+import java.util.Locale;
+
+public class SiegeWarBattleCommanderUtil {
+    
+    /**
+     * Assign the siege commanders for this battle session
+     * To qualify, a player must be:
+     * - Online
+     * - In the siege zone
+     * - Nation king or general
+     * - An official siege participant
+     */
+    public static void assignBattleCommanders() {
+        if(!SiegeWarSettings.isBattleCommandersEnabled()) {
+            return;
+        }
+        Player attackingCommander = null;
+        Player defendingCommander = null;
+        String attackingCommanderName;
+        String defendingCommanderName;
+        for(Siege siege: SiegeController.getSieges()) {
+            if(siege.getStatus().isActive()) {
+                //Wipe the current commanders
+                siege.setAttackingCommander(null);
+                siege.setDefendingCommander(null);
+                //Find the commanders for the siege
+                for(Player player: Bukkit.getOnlinePlayers()) {
+                    if (SiegeWarDistanceUtil.isPlayerRegisteredToActiveSiegeZone(player)
+                            && player.hasPermission(SiegeWarPermissionNodes.SIEGEWAR_NATION_SIEGE_STARTCONQUESTSIEGE.getNode()))
+                    {
+                        if(SiegeSide.getPlayerSiegeSide(siege,player) == SiegeSide.ATTACKERS) {
+                            if(attackingCommander == null) {
+                                attackingCommander = player;
+                                siege.setAttackingCommander(TownyAPI.getInstance().getResident(player));
+                            }
+                        } else if (SiegeSide.getPlayerSiegeSide(siege,player) == SiegeSide.DEFENDERS) {
+                            if(defendingCommander == null) {
+                                defendingCommander = player;
+                                siege.setDefendingCommander(TownyAPI.getInstance().getResident(player));
+                            }
+                        }
+                    }
+                }
+
+                //Display the commanders to all participants
+                attackingCommanderName = attackingCommander == null ? Translatable.of("no_battle_commander").translate(Locale.ROOT) : attackingCommander.getName();
+                defendingCommanderName = defendingCommander == null ? Translatable.of("no_battle_commander").translate(Locale.ROOT) : defendingCommander.getName();
```

Comment:
Have you tested what Towny does when you use Locale.ROOT?

One way to bypass that is to use Translation.of("no_battle_commander") which will use Towny's default locale.